doc/languages-frameworks/javascript.section.md

```diff
@@ -310,6 +310,64 @@ See `node2nix` [docs](https://github.com/svanderburg/node2nix) for more info.
 - `node2nix` has some [bugs](https://github.com/svanderburg/node2nix/issues/238) related to working with lock files from npm distributed with `nodejs_16`.
 - `node2nix` does not like missing packages from npm. If you see something like `Cannot resolve version: vue-loader-v16@undefined` then you might want to try another tool. The package might have been pulled off of npm.
 
+### pnpm {#javascript-pnpm}
+
+Pnpm is available as the top-level package `pnpm`. Additionally, there are variants pinned to certain major versions, like `pnpm_8` and `pnpm_9`.
+
+When packaging an application that includes a `pnpm-lock.yaml`, you need to fetch the pnpm store for that project using a fixed-output-derivation. The `pnpm` derivations come with a `fetchDeps` function and a `configHook` setup hook respectively.
+
+A package containing a `package.json` and a `pnpm-lock.yaml` can be packaged like this:
+
+```nix
+# pkgs.callPackage ... { pnpm = pnpm_9; }
+{ stdenv, nodejs, pnpm }:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "foo";
+  version = "0-unstable-1980-01-01";
+
+  src = ...;
+
+  nativeBuildInputs = [nodejs pnpm.configHook];
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "...";
+  };
+})
+```
+
+NOTE: It is highly recommended to use a pinned version of pnpm (i.e. `pnpm_8` or `pnpm_9`), to increase future reproducibility.
+
+In case you are patching `package.json` or `pnpm-lock.yaml`, make sure to pass `finalAttrs.patches` to the function as well.
```

Comment:
Maybe this will be a bit clearer:

```suggestion
In case you are patching `package.json` or `pnpm-lock.yaml`, make sure to `inherit (finalAttrs) patches` when calling `fetchDeps`.
```